### PR TITLE
#1018 Feedback Box

### PR DESCRIPTION
**Pull request by LisaJD**
Fixed mobile feedback button not sliding, moved js out of feedback.html to static folder, fixed indentation

### CODE REVIEW

File: eahub/feedback/apps.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from django import apps
+
+
+class FeedbackConfig(apps.AppConfig):
+    name = "eahub.feedback"
+    verbose_name = "Feedback"
```
Comment (viktor-yunenko) [inline eahub/feedback/apps.py:6]: todo remove

File: eahub/feedback/forms.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+class FeedbackForm(forms.ModelForm):
+    class Meta:
+        model = Feedback
+        fields = ("message", "email", "page_url")
+        widgets = {
+            "message": forms.Textarea(
+                attrs={"rows": 3, "maxlength": 5000, "required": "true"}
+            )
+        }
+        labels = {"message": ("Feedback"), "email": ("Email address (optional)")}
```
Comment (viktor-yunenko) [inline eahub/feedback/forms.py:15]: todo replace with createview

File: eahub/feedback/views.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+def ajax_post_view(request):
+    model = Feedback
+    form_class = FeedbackForm
+    if request.method == "POST":
+        form = FeedbackForm(request.POST)
+        if form.is_valid():
+            form.save()
+            return HttpResponse(status=200)
+        else:
+            return HttpResponse(status=418)
```
Comment (viktor-yunenko) [inline eahub/feedback/views.py:16]: todo replace with createview

File: eahub/base/static/components/feedback.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  var m = document.getElementById("feedbackModal")
+  m.classList.toggle("show");
+  var element = document.getElementById("feedback");
+  element.classList.toggle("active");
+  document.getElementById("overlay").classList.toggle("active");    
+}
+document.getElementById('page_url').value = window.location.href;
+
+async function ajaxsubmit(){
+  var form = document.getElementById('form');
```
Comment (viktor-yunenko) [inline eahub/base/static/components/feedback.js]: I recommend to use only `const` and `let`, var has been depracated 👌

File: eahub/templates/components/feedback.html [hunk 5]
```diff
@@ -0,0 +1,50 @@
+{% block feedback %}
+{% load static %}
+
+  
+  <!-- Button trigger modal -->
+  <button type="button" class="btn feedbackbtn" data-bs-toggle="modal" data-bs-target="#feedbackModal">
+    Feedback?
```
Comment (sebgrebe) [inline eahub/templates/components/feedback.html:7]: When I click on the button, nothing happens (Desktop, Firefox)

**Review by sebgrebe — CHANGES_REQUESTED** [review]

File: eahub/templates/components/feedback.html (hunk 5 above)
Comment (LisaJD) [reply eahub/templates/components/feedback.html:7]: That's odd... It works for me and I tested in Safari, Chrome and Firefox on desktop and small screens:

image

Is this the correct way to link the javascript?
`<script src="{% static 'components/feedback.js' %}" type="text/javascript"></script>`

Comment (sebgrebe) [reply eahub/templates/components/feedback.html:7]: Hm, it's not working for me. Maybe you didn't push your latest commits?

Comment (sebgrebe) [reply eahub/templates/components/feedback.html:7]: figured it out: had to rerun ```npm ci```.

**Review by sebgrebe — APPROVED** [review]
Approved, created new MR to add implementation of minir code review comments: https://github.com/rtcharity/eahub.org/pull/1225

**Comment by sebgrebe** [comment]
Closing in lieu of new MR, which implements the code review comment from @victor-yunenko: https://github.com/rtcharity/eahub.org/pull/1225

**Pull request closed without merging**